src/actions_ls.rs

```diff
@@ -212,7 +228,7 @@ impl ActionHandler {
         let vfs: &Vfs = &self.vfs;
         let result: Vec<CompletionItem> = panic::catch_unwind(move || {
             let pos = adjust_vscode_pos_for_racer(params.position);
-            let file_path = &params.textDocument.file_name();
+            let file_path = &uri_string_to_file_name(&params.text_document.uri);
```

Comment:
Perhaps a macro that works on anything with a uri field (or even a textDocument field, depending on what the other uses are like)? Not the end of the world if not. I guess it is not much code dup, it just looks like a bit more complicated than the old version.